python_as4.py

```diff
@@ -0,0 +1,138 @@
+class BankAccount:                  #Parent class for bank account
+    def __init__(self,otp):
+        self.balance=1000
+        self.otp=otp
+
+    #Method to check the otp
+    def check_otp(self):                
+        otp_check=int(input("Enter your otp number :"))
+        if self.otp==otp_check:
+            return True
+
+    def wrong(self):
+        print('OTP is wrong . Please re-enter the OTP')
+
+    #Method to check the balance amount
+    def check_balance(self):
+        if self.check_otp():
+            print(f'Your balance is {self.balance}')
+        else:
+            self.wrong()
+
+# Child class for the BankAccount
+class BankActions(BankAccount):
+
+    #Method to deposit the money
+    def deposit(self):
+        if self.check_otp():
+            amount=int(input('Enter the amount you need to deposit :'))
+            self.balance+=amount
+        else:
+            self.wrong()
+
+    #Method to withdraw the money
+    def withdraw(self):
+        if self.check_otp():
+            amount=int(input('Enter the amount you need to withdraw :'))
+            if (amount>self.balance):
+                print("Insufficient balance!")
+            else:
+                self.balance-=amount
+        else:
+            self.wrong()
+
+#importing the random module
+import random
+flag=1
+
+print("Welcome to the ATM ")
+
+# Generating the 4 digits random number
+otp=random.randint(1000,9999)
+print('otp for bank operations is :',otp)
+
+#Object creation
+customer=BankActions(otp)
+while(flag==1):
```

Comment:
Put the business logic inside any of the function

Reply:
ok sir I have added the business logics